src/managers/config.py

```diff
@@ -464,22 +321,22 @@ def inter_broker_protocol_version(self) -> str:
             String with the `major.minor` version
         """
         # Remove patch number from full vervion.
-        major_minor = self.charm.upgrade.current_version.split(".", maxsplit=2)
+        major_minor = self.current_version.split(".", maxsplit=2)
         return ".".join(major_minor[:2])
 
     @property
-    def rack_properties(self) -> List[str]:
+    def rack_properties(self) -> list[str]:
         """Builds all properties related to rack awareness configuration.
 
         Returns:
             List of properties to be set
         """
         # TODO: not sure if we should make this an instance attribute like the other paths
-        rack_path = f"{self.charm.snap.CONF_PATH}/rack.properties"
-        return safe_get_file(rack_path) or []
+        rack_path = f"{self.workload.paths.conf_path}/rack.properties"
```

Comment:
**praise** Ideally, this should have gone in the path, but I totally agree that - given the narrow and "hidden" scope of this feature - I would not use explicitely refer to this in the Path object